### ISSUE TITLE
Manual Form Submit Action with Updated Values

### ISSUE THREAD

**Issue by rpf3** [sveltejs/kit#7307]
### Describe the bug

I am working with a component that only dispatches events but I want to submit a form action when the event is handled.  To do this, I thought I would create a form with a hidden input.  When I handle the event from the component, I take the values from the event details and update my form input.  Then I submit the form by calling `.click` on a bound submit button inside the form.  What I am seeing though is that the state that Svelte is reporting does not match the state being sent to the form action.

### Reproduction

https://github.com/rpf3/form-action-submit-bug

In the example above, I provide 3 buttons.  Two of them increment and decrement a counter.  The value of the hidden form input is set to the value of this counter.  The "submit" button has a click event handler that will double the value and then fire the `.click` event on the hidden form button.  When this happens, I'd expect the form data in the server action to reflect the new doubled value.  However it does not; it is set to the original value before the click event handler runs.

I have come up with a [workaround](https://github.com/rpf3/form-action-submit-bug/blob/main/src/routes/%2Bpage.svelte#L15-L22) which is to split the code into two click event handlers: the first one updates the value and the second one submits the form.  This leads me to believe there is something in the Svelte internals that I haven't found yet which is syncing the state of the DOM only after the click event fires.

### System Info

```shell
System:
  OS: Linux 4.19 Ubuntu 20.04.4 LTS (Focal Fossa)
  CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  Memory: 1.10 GB / 3.84 GB
  Container: Yes
  Shell: 5.0.17 - /bin/bash
Binaries:
  Node: 16.16.0 - ~/.asdf/installs/nodejs/16.16.0/bin/node
  npm: 8.11.0 - ~/.asdf/plugins/nodejs/shims/npm
npmPackages:
  @sveltejs/adapter-auto: next => 1.0.0-next.83
  @sveltejs/kit: next => 1.0.0-next.517
  svelte: ^3.44.0 => 3.52.0
  vite: ^3.1.0 => 3.1.8
```

### Severity

annoyance

### Additional Information

I think this is a strange use case because I don't have direct control over the component and thus need to do this intercept on the dispatched event.

**Comment by dummdidumm** [comment]
@mustofa-id is right, this has to do with the state not being reflected back to the DOM yet, `await tick()` should do the trick, therefore closing the issue.

Having said that, I'm wondering why you are not calling the action on the server directly through `fetch`, or why you don't use `+server.js` since this won't work without JavaScript so there's no reason to use form actions.

**Comment by rpf3** [comment]
@mustofa-id thank you, that was exactly the solution.

@dummdidumm you raise an even better question.  I started with the intention building the feature to work without JavaScript which is why I began the path of form actions.  However, I guess when I came up to the limitation of the component I am using I didn't stop and reassess how that would affect the plan.  In any case, thank you both for your help!
